### ISSUE TITLE
Make gmaps DI (depedency injection) friendly library

### ISSUE THREAD

**Issue by ericnewton76** [ericnewton76/gmaps-api-net#95]
I believe this requires some refactoring that might slightly modify the public interfaces, but for a good cause: depedency injection and inversion of control concepts.

For instance, instead of 
```
var request = new GeocodingRequest();
request.Address = "1600 Amphitheatre Parkway";
request.Sensor = false;
var response = new GeocodingService().GetResponse(request);
```

DI friendly public interfaces would make the following code, which is much more DI friendly itself:
```
public class Consumer 
{
  private IGeocodingService _GeocodingService;
  public Consumer(IGeocodingService GeocodingService) { ... }
   
  public void DoWork()
  {
    IGeocodingRequest request = _GeocodingService.CreateRequest();
    request.Address = "1600 Amphitheatre Parkway";
    var response = _GeocodingService.GetResponse(request);
  }
}
```

**Comment by richardthombs** [comment]
I think it's a great idea to define an interface for each service so that it can be easily mocked for consumers while testing. Something simple like:

```c#
public interface IGeocodingService
{
  GeocodingResponse GetResponse(GeocodingRequest request);
}
```

I struggle to imagine a scenario where somebody would want to swap out Gmaps with something else but still use our interfaces though, and isn't that what you'd use DI for?

**Comment by ericnewton76** [comment]
Not exactly, it's more for the consumers of this library to be able to mock the gmaps away from their unit/integration testing

**Comment by richardthombs** [comment]
Why would the request and response objects need interfaces?

**Comment by ericnewton76** [comment]
for the same mocking reasons.  in DI preferably everything is an interface, makes it much easier to mock

And realistically the request and response objects already have an interface... the implicit one... =D

**Comment by ericnewton76** [comment]
btw, here's what I'm considering:
https://stackoverflow.com/a/2047657/323456
